homeassistant/components/tplink_omada/switch.py

```diff
@@ -110,3 +175,78 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         self.port_details = self.coordinator.data[self.port_id]
         self._refresh_state()
+
+
+@dataclass
+class GatewayPortSwitchConfig:
```

Comment:
Please use an entity description instead. See comment above.

Reply:
Done. I left the PoE port switch entities alone for now, but I will refactor these eventually too.